panther_analysis_tool/main.py

```diff
@@ -398,6 +399,94 @@ def upload_analysis(args: argparse.Namespace) -> Tuple[int, str]:
     return 0, ""
 
 
+def confirm_analysis_exists(args: argparse.Namespace, analysis_id_list: list) -> list:
+    validation_payload = {"listDetections": {"ids": analysis_id_list, "fields": ["id"]}}
+    client = get_client(args.aws_profile, "lambda")
+
+    validation = client.invoke(
+        FunctionName="panther-analysis-api",
+        InvocationType="RequestResponse",
+        LogType="None",
+        Payload=json.dumps(validation_payload),
+    )
+
+    api_response = json.loads(validation["Payload"].read().decode("utf-8"))
+    json_response = api_response["body"]
+
+    # The mock will result in a dict, the actual api returns a string.
+    # The following block deals with that deviation
+    if isinstance(json_response, dict):
+        json_output = json_response
+    # Normal Execution will hit the else clause
+    else:
+        json_output = json.loads(json_response)
```

Comment:
As mentioned in the comment, the api returns a string as the value for `body` where as the mock returns a dict. I tried a lot of iterations of this to make handling the deviation as clean as possible and this is where I landed ( with a significant amount of help from @wey-chiang ). All the typecasting and string foo was quite a bit uglier.

All that said, if there is a better way I'm very open to it. Tagging the Python gurus @lindsey-w @georgepsarakis  @wey-chiang @nhakmiller